### PR TITLE
chore(web): fixes implicit-this cases 🔩

### PR DESCRIPTION
**Pull request by jahorton**
Addresses part of #11473, correcting all remaining cases caught by enforcing `"noImplicitThis" across all Keyman Engine for Web related code.

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by mcdurdin** [comment]
Can you include a reference for why this is being done? (It's part of a larger task, right?)

File: web/src/app/webview/src/oskConfiguration.ts [hunk 1]
```diff
@@ -13,13 +13,9 @@ export function setupEmbeddedListeners(engine: KeymanEngine, osk: OSKView) {
         engine.showKeyboardList();                       // Is connected to VisualKeyboard event: globeKey
       }
     } else if(osk.vkbd) {
-      if(osk.vkbd.menuEvent) {
-        this.highlightKey(osk.vkbd.menuEvent, false);
-      }
       if(typeof(engine.menuKeyUp) == 'function') { // VisualKeyboard event:  globeKey
         engine.menuKeyUp();
       }
-      osk.vkbd.menuEvent = null;
```
Comment (mcdurdin) [inline web/src/app/webview/src/oskConfiguration.ts:22]: This change doesn't seem to belong? Seems like a difference in behaviour.

File: web/src/tsconfig.dom.json [hunk 2]
```diff
@@ -6,7 +6,6 @@
     // TODO: These override ../tsconfig.base.json settings, and so should be removed if possible,
     //       but existing code in web/ breaks some of these settings
     "noImplicitAny": false,
-    "noImplicitThis": false,
```
Comment (jahorton) [inline web/src/tsconfig.dom.json:9]: Why this is being done:  the `// TODO:` above.

File: web/src/engine/element-wrappers/src/utils.ts [hunk 3]
```diff
@@ -24,15 +24,8 @@ export function nestedInstanceOf(Pelem: Event|EventTarget, className: string): b
...
-
-    if(this.instanceof(event.target, 'Window')) {
-      scopedClass = event.target[className];
-    } else if(this.instanceof(event.target, 'Document')) {
-      scopedClass = (event.target as Document).defaultView[className];
-    } else if(this.instanceof(event.target, 'HTMLElement')) {
-      scopedClass = (event.target as HTMLElement).ownerDocument.defaultView[className];
-    }
+    const event = Pelem as Event;
+    return nestedInstanceOf(event.target, className);
```
Comment (mcdurdin) [inline web/src/engine/element-wrappers/src/utils.ts:28]: This also looks like a behavioural change?

**Review by mcdurdin — APPROVED** [review]
LGTM. Sounds like there are some dead code removal PRs coming up before too long as well?

**Review by ermshiperete — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.45-alpha](https://keyman.com/downloads/releases/alpha/18.0.45)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.45-alpha](https://keyman.com/downloads/releases/alpha/18.0.45)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.45-alpha](https://keyman.com/downloads/releases/alpha/18.0.45)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.45-alpha](https://keyman.com/downloads/releases/alpha/18.0.45)

**Pull request merged**